### PR TITLE
Updates to run on 1.0

### PR DESCRIPTION
**Pull request by quinnj**
cc: @nalimilan @ararslan @bkamins

### CODE REVIEW

File: src/abstractdataframe/iteration.jl [hunk 1]
```diff
@@ -19,13 +19,9 @@ which is a view that acts like a one-row DataFrame.
 eachrow(df::AbstractDataFrame) = DFRowIterator(df)
 
 function Base.iterate(itr::DFRowIterator, i=1)
-    if i > size(itr.df, 1)
-        nothing
-    else
-        (DataFrameRow(itr.df, i), i + 1)
-    end
+    i > size(itr.df, 1) && return nothing
+    return (DataFrameRow(itr.df, i), i + 1)
```
Comment (ararslan) [inline src/abstractdataframe/iteration.jl]: This is fine, but isn't it the same as what's already there? Not sure if I'm missing something.

File: src/abstractdataframe/show.jl [hunk 2]
```diff
@@ -1,3 +1,5 @@
+export showall
```
Comment (ararslan) [inline src/abstractdataframe/show.jl]: Was this removed from Base?

Comment (quinnj) [reply src/abstractdataframe/show.jl]: Yes

File: test/dataframe.jl [hunk 3]
```diff
@@ -179,7 +179,7 @@ module TestDataFrame
         @test size(df, 2) == 5
         @test typeof(df[:, 1]) == Vector{Float64}
 
-        df = convert(DataFrame, Matrix{Float64}(I, 10, 5))
+        df = convert(DataFrame, Matrix{Float64}(undef, 10, 5))
```
Comment (ararslan) [inline test/dataframe.jl]: The `I` there was intentional, as previously this was using `eye`.

File: src/abstractdataframe/iteration.jl (hunk 1 above)
Comment (quinnj) [reply src/abstractdataframe/iteration.jl]: It is, just more compact; I actually did this before @bkamins, and I resolved the conflict to my version.

File: test/dataframe.jl (hunk 3 above)
Comment (quinnj) [reply test/dataframe.jl]: Is there any way to get that in 1.0? I just realized we weren't actually testing any `I`-specific values, so 🤷‍♂️

File: test/io.jl [hunk 4]
```diff
@@ -1,28 +1,28 @@
 module TestIO
     using Test, DataFrames, CategoricalArrays, Dates
-    using LaTeXStrings
+    # using LaTeXStrings
```
Comment (ararslan) [inline test/io.jl]: Could you add a comment explaining why this and the corresponding tests are commented out?

File: test/dataframe.jl (hunk 3 above)
Comment (ararslan) [reply test/dataframe.jl]: Yeah `Matrix{Float64}(I, 10, 5)` is how you get it in 1.0, you just need `using LinearAlgebra` to get `I`.

File: test/io.jl (hunk 4 above)
Comment (quinnj) [reply test/io.jl]: Added.

File: src/abstractdataframe/join.jl [hunk 5]
```diff
@@ -202,9 +204,9 @@ function update_row_maps!(left_table::AbstractDataFrame,
     if map_rightonly
         rightonly_orig_ixs = findall(rightonly_mask)
         rightonly_ixs = RowIndexMap(rightonly_orig_ixs,
-                                    collect(length(right_ixs.orig) +
-                                            (leftonly_ixs === nothing ? 0 : length(leftonly_ixs)) +
-                                            (1:length(rightonly_orig_ixs))))
+                                    collect(adjustrange(length(right_ixs.orig) +
+                                            (leftonly_ixs === nothing ? 0 : length(leftonly_ixs)),
```
Comment (nalimilan) [inline src/abstractdataframe/join.jl]: Alignment is incorrect here, the line is very hard to parse.

Also wouldn't `.+` be enough?

File: src/abstractdataframe/show.jl [hunk 6]
```diff
@@ -508,7 +514,7 @@ end
 #'
 #' df = DataFrame(A = 1:3, B = ["x", "y", "z"])
 #' showall(stdout, df, false, :Row, true)
-function Base.showall(io::IO,
+function showall(io::IO,
                       df::AbstractDataFrame,
```
Comment (nalimilan) [inline src/abstractdataframe/show.jl]: Incorrect indentation. Same below.
